### DISCUSSION TITLE
Can `FormattedDateTime` implement `Yokeable` ?

### DISCUSSION THREAD

**Discussion by Baptistemontan** [unicode-org/icu4x#5390] [Q&A]
I need a way to move a `FormattedDateTime` with it's formatter wrapped in a `Rc<DateTimeFormatter>`, I thought I could use `yoke` for that but it seams like `FormattedDateTime` is invariant due to the underlyings `ZeroMap`, is there a way to archieve what I want ?

**Comment by Manishearth** [comment]
Can you expand on your reason to wish to hold on to the FormattedDateTime? it's really intended to be a short lived type.

In theory it's not invariant: ZeroMaps hit a limitation in the compiler, so in theory it could implement Yokeable with prove_covariance_manually, but in practice I'm not convinced it's worth it since I don't see a strong use case.

**Reply by sffc** [reply]
`FormattedDateTime` is designed as a temporary, short-lived object to be converted immediately into a storage type: usually a `String` but possibly a `Vec` of parts (if you need formatToParts).
